pkg/config/deployment.go

```diff
@@ -523,3 +531,76 @@ func (n *NotificationMention) Validate() error {
 	}
 	return fmt.Errorf("event %q is incorrect as NotificationEventType", n.Event)
 }
+
+// PostSync provides all configurations to be used once the current deployment
+// is triggered successfully.
+type PostSync struct {
+	DeploymentChain *DeploymentChain `json:"chain"`
+}
+
+func (p *PostSync) Validate() error {
+	if dc := p.DeploymentChain; dc != nil {
+		return dc.Validate()
+	}
+	return nil
+}
+
+// DeploymentChain provides all configurations to used to trigger a chain of deployments.
+type DeploymentChain struct {
+	// Nodes provides list of DeploymentChainNodes which contain filters to be used
+	// to find applications to deploy as chain node. It's required to not empty.
+	Nodes []*DeploymentChainNode `json:"applications"`
+	// Conditions provides configuration used to determine should the piped in charge in
+	// the first applications in the chain trigger a whole new deployment chain or not.
+	// If this field is not set, always trigger a whole new deployment chain when the current
+	// application is triggered.
+	Conditions *DeploymentChainTriggerCondition `json:"conditions,omitempty"`
+}
+
+func (dc *DeploymentChain) Validate() error {
+	if len(dc.Nodes) == 0 {
+		return fmt.Errorf("missing specified applications that will be triggered on this chain of deployment")
+	}
+
+	for _, n := range dc.Nodes {
+		if err := n.Validate(); err != nil {
+			return err
+		}
+	}
+
+	if cc := dc.Conditions; cc != nil {
+		if err := cc.Validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DeploymentChainNode provides filters used to find the right applications to trigger
+// as a part of the deployment chain.
+type DeploymentChainNode struct {
+	AppName   string            `json:"name"`
+	AppKind   string            `json:"kind"`
+	AppLabels map[string]string `json:"labels"`
+}
+
+func (n *DeploymentChainNode) Validate() error {
+	hasFilterCond := n.AppName != "" || n.AppKind != "" || len(n.AppLabels) != 0
+
+	if !hasFilterCond {
+		return fmt.Errorf("at least one of \"name\", \"kind\" or \"labels\" must be set to find applications to deploy")
+	}
+	return nil
+}
+
+type DeploymentChainTriggerCondition struct {
+	CommitPrefix string `json:"commitPrefix"`
+}
+
+func (c *DeploymentChainTriggerCondition) Validate() error {
+	if c.CommitPrefix == "" {
+		return fmt.Errorf("missing commitPrefix configration as deployment chain trigger condition")
+	}
```

Comment:
nit: In case that the conditions are going to be increased, it would be good to make it clear in some way that `CommitPrefix` is not mandatory, but that at least one condition needs to be specified. For example:

```suggestion
	hasCond := c.CommitPrefix != "" 
	if !hasCond {
		return fmt.Errorf("missing commitPrefix configration as deployment chain trigger condition")
	}
```